components/crypto/src/rand.rs

```diff
@@ -0,0 +1,17 @@
+// Copyright 2023 TiKV Project Authors. Licensed under Apache-2.0.
+
+//! Utilities for cryptographically strong random number generation.
+
+use openssl::{error::ErrorStack, rand};
+
+/// Return a random u64.
+pub fn rand_u64() -> Result<u64, ErrorStack> {
+    let mut rand_id = [0u8; 8];
+    rand::rand_bytes(&mut rand_id)?;
+    Ok(u64::from_ne_bytes(rand_id))
+}
+
+/// Fill buffer with cryptographically strong pseudo-random bytes.
+pub fn rand_bytes(buf: &mut [u8]) -> Result<(), ErrorStack> {
+    rand::rand_bytes(buf)
+}
```

Comment:
```suggestion
/// Fill buffer with cryptographically strong pseudo-random bytes.
pub fn rand_bytes(buf: &mut [u8]) -> Result<(), ErrorStack> {
    rand::rand_bytes(buf)
}

/// Return a random u64.
pub fn rand_u64() -> Result<u64, ErrorStack> {
    let mut rand_id = [0u8; 8];
    rand_bytes(&mut rand_id)?;
    Ok(u64::from_ne_bytes(rand_id))
}
```

Reply:
Why not directly use `crate::rand_bytes()` in the implementation of `rand_u64()` to unify the calling of generating random bytes?